app/models/vendor.rb

```diff
@@ -0,0 +1,47 @@
+class Vendor < ActiveRecord::Base
+  belongs_to :market
+  has_many :sales
+  has_many :products
+
+  validates_presence_of :name, :number_of_employees, :market_id
+  validates_numericality_of :number_of_employees, :market_id
+
+  def recent_sales_first(how_many = sales.count)
+    sorted_sales = sales.sort_by { |sale| sale.purchase_time }
+    sorted_sales.reverse!
+
+    return sorted_sales[0, how_many]
+  end
+
+  def total_sales
+    amounts = sales.map { |sale| sale.amount }
+    total = amounts.reduce { |sum, n| sum + n }
```

Comment:
[`sum`](http://apidock.com/rails/Enumerable/sum) would be a great option here instead of reduce.